homeassistant/components/dwd_weather_warnings/coordinator.py

```diff
@@ -23,4 +24,11 @@ def __init__(self, hass: HomeAssistant, api: DwdWeatherWarningsAPI) -> None:
 
     async def _async_update_data(self) -> None:
         """Get the latest data from the DWD Weather Warnings API."""
-        await self.hass.async_add_executor_job(self.api.update)
+        if self.config_entry is not None:
+            if gps_tracker := self.config_entry.data.get(CONF_GPS_TRACKER, None):
```

Comment:
```suggestion
            if gps_tracker := self.config_entry.data.get(CONF_GPS_TRACKER):
```

Reply:
Updated in 30d0fe3ed0de3df43b59ffdb03292116aa3dc46a